cmp/grid/GridModel.js

```diff
@@ -443,22 +449,39 @@ export class GridModel extends HoistModel {
      * If multiple records are selected, scroll to the first record and then the last. This will do
      * the minimum scrolling necessary to display the start of the selection and as much as
      * possible of the rest.
+     *
+     * Any selected records that are hidden because their parent rows are collapsed will first
+     * be revealed by expanding their parent rows.
+     *
+     * This method imposes a minimal delay to allow the underlying grid implementation to
+     * render all pending data changes.
      */
-    ensureSelectionVisible() {
-        const {records} = this.selModel,
-            {agApi} = this;
+    async ensureSelectionVisibleAsync() {
```

Comment:
The changes to ensureSelectionVisibleAsync in this branch do not fix an issue tested 
in this toolbox branch: https://github.com/xh/toolbox/tree/gridEnsureSelectionVisible2
on this page http://localhost:3000/admin/tests/grid by clicking on the "Select and Go" button.

That button changes the ag-grid row selection via the agrid api and then calls ensureSelectionVisible.

In this branch the new ensureSelectionVisibleAsync does not wait for the store's selection count to catch up to the grid's selection count.  So, it finds no selections.

This "not checking Grid rows selected may be a deliberate choice.  Perhaps to try to guide the user to use Hoist functions to make selections.  But then maybe the method docs should point this out, or the name might be changed to indicate this, ie: ensureRecordSelectionVisibleAsync